### DISCUSSION TITLE
RTP Packet Lost

### DISCUSSION THREAD

**Discussion by delphikit** [bluenviron/mediamtx#1386] [Questions]
Hi. I would like to understand what it means when the logs report multiple lines of RTP Packet Lost. I am using FFmpeg to output an RTSP stream to rtsp simple server and then using a custom app to consume the rtsp from rtsp simple server. What does the packet lost refer to? Is it between ffmpeg and rtsp simple server or between rtsp simple server and my custom app? Thanks!

**Comment by aler9** [comment]
Hello, in order to find out where the packet loss is, you may take a look at the log line that says `RTP Packet lost`. If the line is associated with the session that is publishing the stream, then the loss is between FFmpeg and the server:

```
2023/02/04 15:24:31 INF [RTSP] [session 27153e0a] is publishing to path 'stream', with UDP, 1 track (H264)
2023/02/04 15:24:31 INF [RTSP] [session 27153e0a] 15 RTP packet(s) lost
```

If the line is associated with a session the is reading the stream, then the loss is between the server and your custom app:

```
2023/02/04 15:25:52 INF [RTSP] [session 4052528b] is reading from path 'stream', with UDP, 1 track (H264)
2023/02/04 15:25:52 WAR [RTSP] [session 4052528b] 20 RTP packet(s) lost
```

In order to avoid packet losses, try decreasing the stream bitrate or use one of the techniques listed here:

https://github.com/aler9/rtsp-simple-server#corrupted-frames

[a comment by an automated account is not shown]
